### PR TITLE
Fuzz the evaluator.

### PR DESCRIPTION
**Pull request by a2aaron**
Adds the rust fuzzer and fuzzes `eval_beat`

### CODE REVIEW

File: fuzz/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+path = ".."
+[dependencies.libfuzzer-sys]
+git = "https://github.com/rust-fuzz/libfuzzer-sys.git"
+
+# Prevent this from interfering with workspaces
+[workspace]
+members = ["."]
+
+[[bin]]
+name = "fuzz_target_1"
```
Comment (porglezomp) [inline fuzz/Cargo.toml]: Rename the fuzz target

File: fuzz/fuzz_targets/fuzz_target_1.rs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#![no_main]
+#[macro_use] extern crate libfuzzer_sys;
+extern crate bytebeat;
+
+fuzz_target!(|data: &[u8]| {
+    if let Ok(s) = std::str::from_utf8(data) {
+        if let Ok(program) = bytebeat::parse_beat(s) {
+            if let Ok(program) = bytebeat::compile(program) {
+                let _ = bytebeat::eval_beat(&program, 0.0);
```
Comment (porglezomp) [inline fuzz/fuzz_targets/fuzz_target_1.rs]: We want to evaluate our program over different data. This current setup won't catch any bugs that depend on the value of `t`, so we need the fuzzer to be able to control `t`. We could probably get this by splitting off the first value of the data as a number.

**Review by porglezomp — DISMISSED** [review]

**Comment by a2aaron** [comment]
~~self note: Latest commit `d04f18e` will panic due to data being zero length occasionally.~~

fixed

**Review by porglezomp — APPROVED** [review]
Looks good!

**Pull request merged**